### PR TITLE
KYLIN-3926 set sourceRecordCount when updating statistics

### PR DESCRIPTION
**Pull request by kyotoYaho**

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

File: engine-mr/src/main/java/org/apache/kylin/engine/mr/steps/MergeStatisticsStep.java [hunk 1]
```diff
@@ -120,8 +129,11 @@ protected ExecuteResult doWork(ExecutableContext context) throws ExecuteExceptio
                         tempFile.delete();
                 }
             }
+            sourceRecordCount *= effectiveTimeRange == 0 ? 0 : newSegment.getTSRange().duration() / effectiveTimeRange;
```
Comment (nichunen) [inline engine-mr/src/main/java/org/apache/kylin/engine/mr/steps/MergeStatisticsStep.java]: why not convert to double to calculate the accurate data?

File: engine-mr/src/main/java/org/apache/kylin/engine/mr/common/CubeStatsWriter.java [hunk 2]
```diff
@@ -43,6 +43,11 @@ public static void writeCuboidStatistics(Configuration conf, Path outputPath, //
         writeCuboidStatistics(conf, outputPath, cuboidHLLMap, samplingPercentage, 0, 0, 0);
     }
 
+    public static void writeCuboidStatistics(Configuration conf, Path outputPath, //
+            Map<Long, HLLCounter> cuboidHLLMap, int samplingPercentage, long sourceRecordCoun) throws IOException {
+        writeCuboidStatistics(conf, outputPath, cuboidHLLMap, samplingPercentage, 0, 0, sourceRecordCoun);
```
[the remark this answers is not in the document]
Comment (kyotoYaho) [reply engine-mr/src/main/java/org/apache/kylin/engine/mr/common/CubeStatsWriter.java]: Just copied from the existing parameter name. However, it's better to change it from sourceRecordCoun to sourceRecordCount.

File: engine-mr/src/main/java/org/apache/kylin/engine/mr/steps/MergeStatisticsStep.java (hunk 1 above)
Comment (kyotoYaho) [reply engine-mr/src/main/java/org/apache/kylin/engine/mr/steps/MergeStatisticsStep.java]: It's not necessary to do so. Row count itself is an approximate one. And this data is set to long type at other place and we shouldn't break the compatibility.

Comment (kyotoYaho) [reply engine-mr/src/main/java/org/apache/kylin/engine/mr/steps/MergeStatisticsStep.java]: For details, please check http://issues.apache.org/jira/browse/KYLIN-3453

**Comment by nichunen** [comment]
Should SparkMergingDictionary MergeDictionaryMapper also be changed?

**Comment by nichunen** [comment]
@kyotoYaho Hi, would you please update the pr?

**Pull request merged**